### PR TITLE
fix: token list fetching performance

### PR DESCRIPTION
**Pull request by zzmp**
Avoids refetching an already-fetched token list

### CODE REVIEW

File: src/lib/hooks/useTokenList/index.ts [hunk 1]
```diff
@@ -14,31 +14,34 @@ import { validateTokens } from './validateTokenList'
...
-        // TODO(zzmp): Use network resolver when wallet is not on chainId === 1.
         return resolveENSContentHash(ensName, library)
       }
       throw new Error('Could not construct mainnet ENS resolver')
     },
     [chainId, library]
   )
+
   useEffect(() => {
+    if (tokenList.map) return
```
Comment (JFrankfurt) [inline src/lib/hooks/useTokenList/index.ts:43]: maybe `Array.isArray` instead? checking for method presence feels funky

Comment (JFrankfurt) [reply src/lib/hooks/useTokenList/index.ts:43]: oh, `.map` is not the method, it's a property w/ a `TokenMap` value. THAT's confusing af.

File: src/lib/hooks/useTokenList/index.ts [hunk 2]
```diff
@@ -14,31 +14,34 @@ import { validateTokens } from './validateTokenList'
 
 export const DEFAULT_TOKEN_LIST = 'https://gateway.ipfs.io/ipns/tokens.uniswap.org'
 
-const chainTokenMapAtom = atom<ChainTokenMap | undefined>(undefined)
+const tokenListAtom = atom<{ list?: string | TokenInfo[]; map?: ChainTokenMap }>({})
```
Comment (JFrankfurt) [inline src/lib/hooks/useTokenList/index.ts:17]: pref change `map` to `tokenMap` and `list` to `tokenInfoList`

**Review by JFrankfurt — APPROVED** [review]

File: src/lib/hooks/useTokenList/index.ts (hunk 2 above)
Comment (ianlapham) [reply src/lib/hooks/useTokenList/index.ts:17]: is `list` the url?

**Review by ianlapham — COMMENTED** [review]
Wondering about naming

**Pull request merged**